### PR TITLE
Add runtime validation for empty parser lists in `or()`, `longestMatch()`, `merge()`, and `concat()`

### PR DESCRIPTION
**Pull request by dahlia**
## Summary

These four combinators rely on compile-time arity guards to reject zero-parser calls, but JavaScript callers (or TypeScript callers using `as never` casts) can bypass the type system and call them with no parsers. When this happens, the behavior is inconsistent and undocumented: `or()` and `longestMatch()` produce a generic no-match error, `merge()` produces a different generic no-match error, and `concat()` silently succeeds with `[]`.

This PR adds a runtime `TypeError` at construction time when any of these combinators is called with zero parser arguments. The check runs after argument extraction in *constructs.ts*, so it also catches cases like `or({})` (options only) or `merge("label", {})` (label and options but no parsers).

```typescript
import { or } from "@optique/core/constructs";

// Now throws: TypeError: or() requires at least one parser argument.
(or as any)();
```

Closes https://github.com/dahlia/optique/issues/403

## Test plan

- Added `assert.throws(TypeError)` tests for zero-parser calls to all four combinators in *constructs.test.ts*
- Added edge case tests: options-only for `or()` and `longestMatch()`, label-only / options-only / label+options for `merge()`
- Updated existing arity guard tests that previously called these functions with zero args without expecting a throw
- All tests pass across Deno, Node.js, and Bun (`mise test`)
- Type checking, linting, and format checks pass (`mise check`)

### CODE REVIEW

File: packages/core/src/constructs.test.ts [hunk 1]
```diff
@@ -82,6 +82,20 @@ function asyncStringValue(): ValueParser<"async", string> {
 }
 
 describe("or", () => {
+  it("should throw TypeError when called with no parsers", () => {
+    assert.throws(
+      () => (or as (...args: unknown[]) => unknown)(),
+      TypeError,
+    );
+  });
```
[the remark this answers is not in the document]
Comment (dahlia) [reply packages/core/src/constructs.test.ts:94]: Fixed in 6b4eba9 — all runtime-validation tests now assert the exact error message string via `{ name: "TypeError", message: "..." }`.

File: packages/core/src/constructs.test.ts [hunk 2]
```diff
@@ -10145,10 +10216,10 @@ describe("branch coverage: constructs.ts edge cases", () => {
       assert.deepEqual(asyncCompleted.value, ["scalar-async", undefined]);
     }
 
-    const emptyConcat = (concat as unknown as (...parsers: unknown[]) => {
-      readonly priority: number;
-    })();
-    assert.equal(emptyConcat.priority, 0);
+    assert.throws(
+      () => (concat as unknown as (...parsers: unknown[]) => unknown)(),
```
[the remark this answers is not in the document]
Comment (dahlia) [reply packages/core/src/constructs.test.ts]: Fixed in 5606932 — replaced all `as unknown as (...) => unknown` double casts with `@ts-expect-error` + direct calls, matching the existing convention used elsewhere in the test file.

File: packages/core/src/constructs.test.ts [hunk 3]
```diff
@@ -366,9 +386,14 @@ describe("or", () => {
   });
 
   it("should report type-level arity error for zero parsers", () => {
```
[the remark this answers is not in the document]
Comment (dahlia) [reply packages/core/src/constructs.test.ts]: Fixed in ab013a1 — renamed all four test titles from "should report type-level arity error" to "should throw TypeError".
